src/monmove.cpp

```diff
@@ -387,7 +387,8 @@ void monster::plan()
         }
     } else if( friendly != 0 && !docile ) {
         for( monster &tmp : g->all_monsters() ) {
-            if( tmp.friendly == 0 && seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
+            if( tmp.friendly == 0 && tmp.attitude_to( *this ) == Attitude::HOSTILE &&
+                seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
```

Comment:
```suggestion
            if( tmp.friendly == 0 && ( tmp.attitude_to( *this ) == Attitude::HOSTILE ||
                                                       has_flag( MF_PET_WONT_FOLLOW ) ) &&
                seen_levels.test( tmp.pos().z + OVERMAP_DEPTH ) ) {
```
(This will likely require some formatting.) Also remove the reference to MF_PET_WONT_FOLLOW from 345, which is dealing with non-pets (pets would be friendly!).

Reply:
I removed it from L345, but I believe the right place to check it is L637 (done)

Reply:
If you remove it from L637, how would one get such animals to move when desired?

Reply:
Reverted, not touching MF_PET_WONT_FOLLOW in this PR